src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/TransitiveNoWarnUtils.cs

```diff
@@ -777,43 +748,54 @@ public bool Equals(NodeWarningProperties other)
                     EqualityUtility.DictionaryEquals(PackageSpecific, other.PackageSpecific, (s, o) => EqualityUtility.SetEqualsWithNullCheck(s, o));
             }
 
+            /// <summary>
+            /// Obtain the intersection of this node and the other node
+            /// This method will return the intersection of the warning properties from both sets, not modifying either object.
+            /// </summary>
+            /// <remarks>Null is considered an empty set, which has an empty intersection with anything else.</remarks>
+            /// <param name="other">other node to intersect with</param>
+            /// <returns>intersection between this node and other node</returns>
             public NodeWarningProperties GetIntersect(NodeWarningProperties other)
             {
-                if (other == null || ReferenceEquals(this, other))
+                if (other == null)
+                {
+                    return null;
+                }
+
+                if (ReferenceEquals(this, other))
                 {
-                    return new NodeWarningProperties(ProjectWide, PackageSpecific);
+                    return this;
                 }
 
                 var thisPackages = PackageSpecific;
                 var otherPackages = other.PackageSpecific;
-                var projectWide = Intersect(ProjectWide, other.ProjectWide);
-                Dictionary<string, HashSet<NuGetLogCode>> packages = null;
 
-                if (thisPackages != null && otherPackages == null)
-                {
-                    packages = thisPackages;
-                }
-                else if (thisPackages == null && otherPackages != null)
-                {
-                    packages = otherPackages;
-                }
-                else if (thisPackages != null && otherPackages != null)
+                // null is empty and cannot intersect
+                Dictionary<string, HashSet<NuGetLogCode>> packages = null;
+                if (thisPackages != null && otherPackages != null)
                 {
                     packages = new Dictionary<string, HashSet<NuGetLogCode>>(StringComparer.OrdinalIgnoreCase);
 
-                    var allKeys = thisPackages.Keys.Concat(otherPackages.Keys)
-                        .Distinct(StringComparer.OrdinalIgnoreCase);
-
-                    foreach (var key in allKeys)
+                    foreach (var pair in thisPackages)
```

Comment:
Please use explicit types instead of `var`.

Reply:
Sorry.  Trying to follow https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md rule 10, but I personally avoid var all of the time so it's hard to find the balance.  Also, I think that this one pre-dated my change and I filed to update it.